### PR TITLE
Added some keyboard shortcut

### PR DESCRIPTION
**Pull request by luca-yinxing**

### CODE REVIEW

File: src/main/resources/i18n.properties [hunk 1]
```diff
@@ -37,3 +37,9 @@ gui.assembly.successful=Assembly successful.
 gui.assembly.failed.with.0.errors=Assembly failed with {0} errors:\n\n
 runtime.error.0.at.1=Runtime error: "{0}" at {1}
 internal.error.0=Internal error:{0}
+new.file=New file (Ctrl+N)
+open.file=Open file (Ctrl+O)
+save.file=Save file (Ctrl+S)
+assemble.program=Assemble program (Ctrl+B)
+step.instruction=Step instruction (F11)
+run.program=Run program (F5)
```
Comment (alessandropellegrini) [inline src/main/resources/i18n.properties:45]: The shortcuts are not portable, as there is no `ctrl` on macOS. The best thing is to remove them, as this should be managed in a proper menu that is still missing.

Also, I would group all these with a leading `gui.` to show what these properties are about.

File: src/main/resources/i18n_es.properties [hunk 2]
```diff
@@ -37,4 +37,10 @@ gui.assembly.successful=Assembly successful.
 gui.assembly.failed.with.0.errors=Assembly failed with {0} errors:\n\n
 runtime.error.0.at.1=Runtime error: "{0}" at {1}
 internal.error.0=Internal error:{0}
+new.file=New file (Ctrl+N)
+open.file=Open file (Ctrl+O)
+save.file=Save file (Ctrl+S)
+assemble.program=Assemble program (Ctrl+B)
+step.instruction=Step instruction (F11)
+run.program=Run program (F5)
```
Comment (alessandropellegrini) [inline src/main/resources/i18n_es.properties:45]: See previous comment.

File: src/main/resources/i18n_fr.properties [hunk 3]
```diff
@@ -37,4 +37,11 @@ gui.assembly.successful=Assembly successful.
 gui.assembly.failed.with.0.errors=Assembly failed with {0} errors:\n\n
 runtime.error.0.at.1=Runtime error: "{0}" at {1}
 internal.error.0=Internal error:{0}
+new.file=New file (Ctrl+N)
+open.file=Open file (Ctrl+O)
+save.file=Save file (Ctrl+S)
+assemble.program=Assemble program (Ctrl+B)
+step.instruction=Step instruction (F11)
+run.program=Run program (F5)
```
Comment (alessandropellegrini) [inline src/main/resources/i18n_fr.properties:45]: See previous comment.

File: src/main/resources/i18n_it.properties [hunk 4]
```diff
@@ -37,4 +37,11 @@ gui.assembly.successful=Assembly successful.
 gui.assembly.failed.with.0.errors=Assembly failed with {0} errors:\n\n
 runtime.error.0.at.1=Runtime error: "{0}" at {1}
 internal.error.0=Internal error:{0}
+new.file=New file (Ctrl+N)
+open.file=Open file (Ctrl+O)
+save.file=Save file (Ctrl+S)
+assemble.program=Assemble program (Ctrl+B)
+step.instruction=Step instruction (F11)
+run.program=Run program (F5)
```
Comment (alessandropellegrini) [inline src/main/resources/i18n_it.properties:45]: See previous comment.

File: src/main/java/it/uniroma2/pellegrini/z64sim/view/MainWindow.java [hunk 5]
```diff
@@ -100,6 +102,46 @@ public void keyTyped(KeyEvent e) {
...
+            e -> this.openFile(),
+            KeyStroke.getKeyStroke(KeyEvent.VK_O, InputEvent.CTRL_DOWN_MASK),
+            JComponent.WHEN_IN_FOCUSED_WINDOW
+        );
+
+        mainPanel.registerKeyboardAction(
+            e -> {
+                SimulatorController.step();
+                },
+            KeyStroke.getKeyStroke(KeyEvent.VK_F11, 0),
```
Comment (alessandropellegrini) [inline src/main/java/it/uniroma2/pellegrini/z64sim/view/MainWindow.java:128]: Maybe `F8` is a more common shortcut?

File: src/main/java/it/uniroma2/pellegrini/z64sim/view/MainWindow.java [hunk 6]
```diff
@@ -100,6 +102,46 @@ public void keyTyped(KeyEvent e) {
         runButton.addActionListener(actionEvent -> {
             SimulatorController.run();
         });
+
+        mainPanel.registerKeyboardAction(
+            e -> this.saveFile(),
+            KeyStroke.getKeyStroke(KeyEvent.VK_S, InputEvent.CTRL_DOWN_MASK),
```
Comment (alessandropellegrini) [inline src/main/java/it/uniroma2/pellegrini/z64sim/view/MainWindow.java:108]: This is not nice to Mac users, as there is no `ctrl` button there.

Java guidelines suggest to rely on the platform's toolkit in `java.awt.Toolkit`, in particular the `getMenuShortcutKeyMaskEx()` method, for this purpose.

**Review by alessandropellegrini — CHANGES_REQUESTED** [review]
Thanks for contributing! I have some concerns about the portability of this PR, that are highlighted in the code review.

File: src/main/resources/i18n.properties (hunk 1 above)
Comment (luca-yinxing) [reply src/main/resources/i18n.properties:45]: I didnt know about that

**Review by alessandropellegrini — APPROVED** [review]

**Pull request merged**